torchao/_models/llama/model.py

```diff
@@ -197,15 +197,17 @@ def forward(self, idx: Tensor, input_pos: Optional[Tensor] = None) -> Tensor:
 
         if input_pos is None: 
             mask = None
+            input_pos = torch.arange(0, idx.shape[1], device=idx.device)
```

Comment:
Hi @HDCharles , I'm not entirely sure if `input_pos` should be initialized this way, please take a look.